### PR TITLE
Upgrade providers to support multi results

### PR DESCRIPTION
**Pull request by ebreton**
Hi @DenisCarriere ,

I will upgrade a few providers to consolidate the new structure of the code.

One at a time :) 
And only the ones with tests. I will leave the others for you since I am not familiar with them.

You can merge the pull request whenever you want. Just leave the branch open and I will create new PRs until I am done  ^^

Cheers
Manu

### CODE REVIEW

**Comment by ebreton** [comment]
okyyy

5 providers done, with google among them, I think the structure is kind of stable.

Feel free to comment or suggest different ways... or merge the PR and release on pypi ! if you feel comfortable with it :)

I will move on another project but I will keep an eye here 👍

File: geocoder/base.py [hunk 1]
```diff
@@ -713,6 +725,18 @@ def __repr__(self):
         else:
             return base_repr.format(u'#%s results' % len(self))
 
+    def _build_headers(self, provider_key, **kwargs):
```
Comment (DenisCarriere) [inline geocoder/base.py]: Nice I like these boilerplate functions :)

**Comment by DenisCarriere** [comment]
@ebreton You're an animal! Great stuff! 🥇

File: geocoder/base.py (hunk 1 above)
Comment (ebreton) [inline geocoder/base.py]: yep, a bit like "hooks" that allows to factorize the main part of the code in base.py, and offers at the same time high flexibility in the cutomization 👍

**Comment by ebreton** [comment]
Happy that you liked the refactoring !

**Comment by DenisCarriere** [comment]
Love it! I definitely promote refactoring, however I've been doing so much NodeJS recently I'm not up to date with Python syntax anymore.

**Pull request merged**